src/tribler/core/components/torrent_checker/torrent_checker/torrentchecker_session.py

```diff
@@ -413,72 +423,59 @@ async def scrape(self):
 
 
 class FakeDHTSession(TrackerSession):
     """
     Fake TrackerSession that manages DHT requests
     """
 
-    def __init__(self, dlmgr, infohash, timeout):
-        super().__init__('DHT', 'DHT', 'DHT', 'DHT', timeout)
-
-        self.infohash = infohash
-        self.dlmgr = dlmgr
+    def __init__(self, download_manager: DownloadManager, timeout: int):
+        super().__init__(DHT, DHT, DHT, DHT, timeout)
 
-    async def cleanup(self):
-        """
-        Cleans the session by cancelling all deferreds and closing sockets.
-        :return: A deferred that fires once the cleanup is done.
-        """
-        await super().shutdown_task_manager()
-        self.infohash_list = None
+        self.download_manager = download_manager
 
-    def add_infohash(self, infohash):
-        """
-        This function adds a infohash to the request list.
-        :param infohash: The infohash to be added.
-        """
-        self.infohash = infohash
-
-    async def connect_to_tracker(self):
-        """
-        Fakely connects to a tracker.
-        :return: A deferred that fires with the health information.
-        """
-        metainfo = await self.dlmgr.get_metainfo(self.infohash, timeout=self.timeout)
-        if not metainfo:
-            raise RuntimeError("Metainfo lookup error")
+    async def connect_to_tracker(self) -> TrackerResponse:
+        responses = []
+        for infohash in self.infohash_list:
+            metainfo = await self.download_manager.get_metainfo(infohash, timeout=self.timeout)
+            health = InfohashHealth(infohash=infohash, seeders=metainfo[b'seeders'], leechers=metainfo[b'leechers'])
+            responses.append(health)
 
-        return {
-            "DHT": [{
-                "infohash": hexlify(self.infohash),
-                "seeders": metainfo[b"seeders"],
-                "leechers": metainfo[b"leechers"]
-            }]
-        }
+        return TrackerResponse(url=DHT, responses=responses)
 
 
 class FakeBep33DHTSession(FakeDHTSession):
     """
     Fake session for a BEP33 lookup.
     """
 
-    async def connect_to_tracker(self):
-        """
-        Fakely connects to a tracker.
-        :return: A deferred that fires with the health information.
-        """
-        try:
-            return await self.dlmgr.dht_health_manager.get_health(self.infohash, timeout=self.timeout)
-        except TimeoutError:
-            self.failed(msg='request timed out')
+    async def connect_to_tracker(self) -> TrackerResponse:
+        tasks = [self.download_manager.dht_health_manager.get_health(infohash, timeout=self.timeout) for infohash in
+                 self.infohash_list]
+        return TrackerResponse(url=DHT, responses=await gather(*tasks))
+
+
+TrackerSessionType = TypeVar('TrackerSessionType', bound=TrackerSession)
+
+
+def create_tracker_session(tracker_url, timeout, proxy, socket_manager) -> TrackerSessionType:
```

Comment:
I think we should not use TypeVar here. TypeVar is useful for specifying a relationship between different arguments. Or, for example, for specifying that the result type should be the same as the type of some argument.

In this function, the result type is not dependent on a type of any argument, so it is possible to directly say that the result type is `TrackerSession`

Reply:
Fixed